src/MOIWrapper.jl

```diff
@@ -1,408 +1,435 @@
+        # Possible bound combinations:
+        #=
+            * No bound (default case)
+            * Interval
+            * EqualTo
+            * LessThan
+            * GreaterThan
+            * LessThan && GreaterThan
+        =#
+        lb, ub = -Inf, Inf  # Default case: free variable
+
+        idx_RG = MOI.ConstraintIndex{MOI.SingleVariable, MOI.Interval{Float64}}(x.value)
+        idx_ET = MOI.ConstraintIndex{MOI.SingleVariable, MOI.EqualTo{Float64}}(x.value)
+        idx_LT = MOI.ConstraintIndex{MOI.SingleVariable, MOI.LessThan{Float64}}(x.value)
+        idx_GT = MOI.ConstraintIndex{MOI.SingleVariable, MOI.GreaterThan{Float64}}(x.value)
+        if MOI.is_valid(src, idx_LT)
+            # Get the (upper) bound
+            s = MOI.get(src, MOI.ConstraintSet(), idx_LT)
+            ub = s.upper
+
+            # Track index of upper-bound constraint
+            mapping.conmap[idx_LT] = MOI.ConstraintIndex{MOI.SingleVariable, MOI.LessThan{Float64}}(j)
+        end
+        if MOI.is_valid(src, idx_GT)
+            # Get the (lower) bound
+            s = MOI.get(src, MOI.ConstraintSet(), idx_GT)
+            lb = s.lower
+
+            # Track index of upper-bound constraint
+            mapping.conmap[idx_GT] = MOI.ConstraintIndex{MOI.SingleVariable, MOI.GreaterThan{Float64}}(j)
         end
-        if sense == Cchar('G')
-            lower[row] = rhs
-            upper[row] = Inf
-        elseif sense == Cchar('L')
-            lower[row] = -Inf
-            upper[row] = rhs
-        elseif sense == Cchar('E')
-            lower[row] = rhs
-            upper[row] = rhs
+        if MOI.is_valid(src, idx_RG)
+            # Get upper and lower bound
+            s = MOI.get(src, MOI.ConstraintSet(), idx_RG)
+            lb, ub = s.lower, s.upper
+
+            # Track index of upper-bound constraint
+            mapping.conmap[idx_RG] = MOI.ConstraintIndex{MOI.SingleVariable, MOI.Interval{Float64}}(j)
         end
+        if MOI.is_valid(src, idx_ET)
+            # Get the bounds
+            s = MOI.get(src, MOI.ConstraintSet(), idx_ET)
+            lb, ub = s.value, s.value
+
+            # Track index of upper-bound constraint
+            mapping.conmap[idx_ET] = MOI.ConstraintIndex{MOI.SingleVariable, MOI.EqualTo{Float64}}(j)
+        end
+
+        Clp.add_column(dest.inner, Cint(0), Int32[], Float64[], lb, ub, 0.0)
     end
-    chg_row_upper(instance.inner, upper)
-    chg_row_lower(instance.inner, lower)
-end
 
-function LQOI.set_linear_objective!(instance::Optimizer, cols::Vector{Int}, coefs::Vector{Float64})
-    objective_coefficients = zeros(Float64, get_num_cols(instance.inner))
-    for (col, coef) in zip(cols, coefs)
-        objective_coefficients[col] += coef
+    # Create linear constraints
+    nrows = 0  # number of rows in the problem
+    for (F, S) in MOI.get(src, MOI.ListOfConstraints())
+        # Check if constraint is a linear constraint
+        (F == MOI.ScalarAffineFunction{Float64}) || continue
```

Comment:
You should throw `MOI.UnsupportedConstraint` if it's not supported, here it will just be silently ignored